### PR TITLE
Normalize dates to Unix timestamps in _seconds_

### PR DESCRIPTION
**Pull request by virtualtam**
TL;DR: Always use units when expressing data. Units are good.

Relates to shaarli/Shaarli#1227

Although most bookmarking services return dates formatted as a Unix epoch (seconds elapsed since `1970-01-01 00:00:00`) or human-readable strings, some services return microtime epochs (microseconds elapsed since `1970-01-01 00:00:00.000000`) WITHOUT using a delimiter for the microseconds part...

This PR adds some guesswork to the date parser so results fall within a resonable range, e.g.:

> `now - 30 years < parsed date < now + 30 years`

This is likely to raise issues in the distant future!

See:
- https://stackoverflow.com/questions/33691428/datetime-with-microseconds
- https://stackoverflow.com/questions/23929145/how-to-test-if-a-given-time-stamp-is-in-seconds-or-milliseconds
- https://stackoverflow.com/questions/539900/google-bookmark-export-date-format
- https://www.wired.com/2010/11/1110mars-climate-observer-report/

### CODE REVIEW

**Review by nodiscc — APPROVED** [review]

File: NetscapeBookmarkParser.php [hunk 1]
```diff
@@ -206,19 +216,58 @@ public function parseString($bookmarkString) {
...
+     * @see https://www.wired.com/2010/11/1110mars-climate-observer-report/
+     *
+     * @param string $epoch     Unix timestamp to normalize
+     *
+     * @return string Unix timestamp in seconds, within the expected range
+     */
+    public function normalizeDate($epoch) {
+        $now = new \DateTime();
+        $range = new \DateInterval('P'.$this->dateDeltaYears.'Y');
+        $maxDate = $now->add($range);
```
Comment (ArthurHoaro) [inline NetscapeBookmarkParser.php]: For simple intervals, you can just use `new \DateTime('+30 years')`.

**Review by ArthurHoaro — APPROVED** [review]

File: NetscapeBookmarkParser.php (hunk 1 above)
Comment (virtualtam) [reply NetscapeBookmarkParser.php]: nice, this is also more user-friendly!
